airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -606,11 +606,22 @@ def delete_objects(self, bucket, keys):
             When ``keys`` is a list, it's supposed to be the list of the
             keys to delete.
         :type keys: str or list
+
+        :return: False on failure to delete at least one batch and True on success.
+        :rtype: bool
         """
         if isinstance(keys, str):
             keys = [keys]
 
-        delete_dict = {"Objects": [{"Key": k} for k in keys]}
-        response = self.get_conn().delete_objects(Bucket=bucket, Delete=delete_dict)
-
-        return response
+        s3 = self.get_conn()
+        batch = 1000
+        for i in range(0, len(keys), batch):
+            try:
+                s3.delete_objects(
+                    Bucket=bucket,
+                    Delete={"Objects": [{"Key": k} for k in keys[i: i + batch]]}
+                )
+            except ClientError as e:
+                self.log.error(e.response["Error"]["Message"])
+                return False
```

Comment:
I am not sure if we should return False on any `ClientError`. Can you be more specific?

Reply:
And maybe then add a test to validate your `try-except` feature :) It should be a noted.

Reply:
I've changed the logic a bit due to failing tests in S3DeleteObjectsOperator.
ClientError is raised in any technical error (e.g. the specified bucked does not exist or the request is throttled).
I've also updated the tests.

Reply:
Ok, I don't see any other method in the code that deals with such technical issues. As long as the task will fail on any exception, it's fine with me.
Removing try-except.